src/components/CalendarContainer.ts

```diff
@@ -0,0 +1,293 @@
+import { Component, createElement } from "react";
+
+import { Alert } from "./Alert";
+import { Calendar, CalendarEvent, View } from "./Calendar";
+
+interface WrapperProps {
+    mxform: mxui.lib.form._FormBase;
+    mxObject: mendix.lib.MxObject;
+}
+
+export interface CalendarContainerProps extends WrapperProps {
+    titleAttribute: string;
+    startAttribute: string;
+    endAttribute: string;
+    defaultView: View;
+    dataSource: DataSource;
+    eventEntity: string;
+    entityConstraint: string;
+    dataSourceMicroflow: string;
+    popup: boolean;
+    selectable: boolean;
+    onClickEvent: OnClickEventOptions;
+    onClickSlotEvent: OnClickEventOptions;
+    eventPage: string;
+    openPageAs: PageLocation;
+    eventMicroflow: string;
+    eventNanoflow: Nanoflow;
+    slotPage: string;
+    slotMicroflow: string;
+    slotNanoflow: Nanoflow;
+}
+
+type DataSource = "XPath" | "microflow";
+type OnClickEventOptions = "doNothing" | "showPage" | "callMicroflow" | "callNanoflow";
+type PageLocation = "content" | "popup" | "modal";
+
+interface CalendarContainerState {
+    alertMessage?: string;
+    events: CalendarEvent[];
+}
+
+interface Nanoflow {
+    nanoflow: object[];
+    paramsSpec: { Progress: string };
+}
+
+export default class CalendarContainer extends Component<CalendarContainerProps, CalendarContainerState> {
+    private subscriptionHandles: number[];
+
+    constructor(props: CalendarContainerProps) {
+        super(props);
+
+        this.subscriptionHandles = [];
+        this.state = {
```

Comment:
Do outside the constructor as instructed before, and remove constructor.

Reply:
Done